### PR TITLE
OSDOCS-10060-16:Documented Support for migrating from SDN to OVN with N

### PR DESCRIPTION
**Pull request by dfitzmau**
Version(s):
4.16

Issue:
[OSDOCS-10060](https://issues.redhat.com/browse/OSDOCS-10060)

Link to docs preview:
* [Considerations for offline migration to the OVN-Kubernetes network plugin](https://80277--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.html#considerations-migrating-ovn-kubernetes-network-provider_migrate-from-openshift-sdn)
* [Migrating to the OVN-Kubernetes network plugin by using the offline migration method](https://80277--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.html#nw-ovn-kubernetes-migration_migrate-from-openshift-sdn)
* [Removing the NodeNetworkConfigurationPolicy (NNCP) custom resource (CR)](https://80277--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.html#removing-nncp-initiating-limited-live-migration_migrate-from-openshift-sdn)

- [x] SME has approved this change on #79807 (which should have targeted 4.16 as the SDN docs will be removed from 4.17)
- [x] QE has approved this change (Ross Brittain or Zhanqi. Messaged them both).

Additional information:
* [NP-1036](https://issues.redhat.com/browse/NP-1036)
* [OPNET-499](https://issues.redhat.com/browse/OPNET-499)
* [Migrating from the OpenShift SDN network plugin](https://docs.openshift.com/container-platform/4.16/networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.html#how-the-migration-process-works_migrate-from-openshift-sdn)
* [Installing the Kubernetes NMState Operator](https://docs.openshift.com/container-platform/4.16/networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.html#installing-the-kubernetes-nmstate-operator-cli)
* [BM Install doc](https://docs.openshift.com/container-platform/4.16/installing/installing_bare_metal/installing-bare-metal-network-customizations.html#creating-manifest-file-customized-br-ex-bridge_installing-bare-metal-network-customizations)

### CODE REVIEW

File: modules/nw-ovn-kubernetes-migration.adoc [hunk 1]
```diff
@@ -7,25 +7,23 @@
...
-* Access to the cluster as a user with the `cluster-admin` role.
-* A recent backup of the etcd database is available.
-* A reboot can be triggered manually for each node.
-* The cluster is in a known good state, without any errors.
-* Before migration to the OVN-Kubernetes plugin, a security group rule must be in place to allow UDP packets on port `6081` for all nodes on all cloud platforms.
-* Before migration to the OVN-Kubernetes plugin, you must either set all timeouts for webhooks to `3` seconds or remove the webhooks.
+* You have a cluster configured with the OpenShift SDN CNI network plugin in the network policy isolation mode.
+* You installed the {oc-first}.
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have a recent backup of the etcd database is available.
```
Comment (tmalove) [inline modules/nw-ovn-kubernetes-migration.adoc:22]:
```suggestion
* You have a recent backup of the etcd database.
```

File: modules/removing-nncp-limited-live-migration.adoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+:_mod-docs-content-type: PROCEDURE
+[id="removing-nncp-limited-live-migration.adoc_{context}"]
+= Removing the `NodeNetworkConfigurationPolicy` (NNCP) custom resource (CR)
+
+The OpenShift SDN plugin allows application of the NodeNetworkConfigurationPolicy (NNCP) custom resource (CR) to the primary interface on a node. The OVN-Kubernetes network plugin does not have this capability.
+
+If you have an NNCP applied to the primary interface, you must delete the NNCP before migrating to the OVN-Kubernetes network plugin. Deleting the NNCP does not remove the configuration from the primary interface, but the Kubernetes-NMState cannot manage this configuration. Instead, the `configure-ovs.sh` shell script manages the primary interface and the configuration attached to it.
+
+.Prerequisties
```
Comment (tmalove) [inline modules/removing-nncp-limited-live-migration.adoc:13]:
```suggestion
.Prerequisites
```

**Review by tmalove — APPROVED** [review]

File: modules/nw-ovn-kubernetes-migration.adoc (hunk 1 above)
Comment (JoeAldinger) [reply modules/nw-ovn-kubernetes-migration.adoc:22]: This is worth changing IMO.

File: modules/removing-nncp-limited-live-migration.adoc (hunk 2 above)
Comment (JoeAldinger) [reply modules/removing-nncp-limited-live-migration.adoc:13]: This needs updated as well.

File: networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc [hunk 3]
```diff
@@ -10,19 +10,46 @@ As a cluster administrator, you can migrate to the OVN-Kubernetes network plugin
...
+// Removing the NodeNetworkConfigurationPolicy (NNCP) custom resource (CR)
+include::modules/removing-nncp-limited-live-migration.adoc[leveloffset=+3]
+
+// Initiating the limited live migration process
 include::modules/initiating-limited-live-migration.adoc[leveloffset=+3]
+
+// Patching OVN-Kubernetes address ranges
 include::modules/patching-ovnk-address-ranges.adoc[leveloffset=+3]
+
+// Checking cluster resources after initiating the limited live migration
```
Comment (JoeAldinger) [inline networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc:46]: Thank you for cleaning up our notes on this fairly complicated migration procedure. Well done!

**Review by JoeAldinger — COMMENTED** [review]
two small things that I _think_ we should fix before merge, please.

File: modules/removing-nncp-limited-live-migration.adoc (hunk 2 above)
Comment (dfitzmau) [reply modules/removing-nncp-limited-live-migration.adoc:13]: Updated

**Pull request merged**